### PR TITLE
Fix --incompatible_disallow_dict_plus

### PR DESCRIPTION
**Pull request by smukherj1**
Also updated a few repositories to pull in fixes to other incompatible
changes that haven't been enabled by default in Bazel yet. More work is
needed in rules_grpc and com_google_protobuf. See #248

### CODE REVIEW

**Comment by smukherj1** [comment]
I think the Bazel version being used by prow might be too old

File: WORKSPACE [hunk 1]
```diff
@@ -146,9 +147,9 @@ go_register_toolchains()
 
 http_archive(
     name = "build_stack_rules_proto",
-    sha256 = "074dbfbeef97dfb2145cdda0bab6d1d51f459e90e0e835284a59b3880bbebd6e",
-    strip_prefix = "rules_proto-137014a36f389cfcb4987a567b7bd23a7a259cf9",
-    urls = ["https://github.com/stackb/rules_proto/archive/137014a36f389cfcb4987a567b7bd23a7a259cf9.tar.gz"],
+    strip_prefix = "rules_proto-f5d6eea6a4528bef3c1d3a44d486b51a214d61c2",
```
Comment (nlopezgi) [inline WORKSPACE]: buildifier will probably complain that sha needs to be before strip_prefix

Comment (smukherj1) [reply WORKSPACE]: Ran buildifier which fixed these

**Pull request merged**